File: LeetCode/50. Pow(x, n).js
/**
 * @param {number} x
 * @param {number} n
 * @return {number}
 */
var myPow = function (x, n) {
  if (n === 0) return 1;
  if (n === 1) return x;
  if (n === -1) return 1 / x;
  if (n % 2 === 0) {
    let t = myPow(x, n / 2);
    return t * t;
  }
  let t = myPow(x, (n - 1) / 2);
  return t * t * x;
};

File: LeetCode/50. Pow(x, n).rs
impl Solution {
  pub fn my_pow(x: f64, n: i32) -> f64 {
      if n==0{
          return 1.0;
      }
      if n==1{
          return x;
      }
      if n==-1{
          return 1.0/x;
      }
      let e=n/2;
      let t=Solution::my_pow(x,e);
      if n%2==0{t*t}else{t*t*if n>0{x}else{1.0/x}}
  }
}